projects/centre_console/inc/button_press.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+// Receives button presses on GPIO, and raises events.
+// Requires interrupts and gpio to be initialized.
+
+#include "gpio.h"
+#include "status.h"
+
+typedef enum {
+  CENTRE_CONSOLE_BUTTON_DRIVE = 0,
+  CENTRE_CONSOLE_BUTTON_REVERSE,
+  CENTRE_CONSOLE_BUTTON_POWER,
+  CENTRE_CONSOLE_BUTTON_NEUTRAL,
+  CENTRE_CONSOLE_BUTTON_HAZARD,
+  CENTRE_CONSOLE_BUTTON_EMERGENCY_STOP,
+  NUM_CENTRE_CONSOLE_BUTTONS
+} CentreConsoleButton;
+
+StatusCode button_press_init(void);
+
+GpioAddress *test_provide_button_addresses(void);
```

Comment:
Weird name imo? I'd have gone with just "get_button_adresses" or smtg probably ...

Reply:
this function gets used in the unit tests to get all the addresses. As they are internal to this module and shouldn't be used by the rest of the modules... the convention is to prefix functions that are ONLY used in tests with `test_`. In general one should try to avoid them, but if you find yourself redefining everything in your tests, then it makes sense to just write a test function like this. :)